drivers/flash/flash_sam0.c

```diff
@@ -51,7 +51,9 @@ struct flash_sam0_data {
 	off_t offset;
 #endif
 
+#if CONFIG_MULTITHREADING
```

Comment:
By convention, we check for whether Kconfig symbols are defined rather than its value.

```suggestion
#ifdef CONFIG_MULTITHREADING
```
OR
```suggestion
#if defined(CONFIG_MULTITHREADING)
```

Reply:
Done as #if defined()